src/gui/toolbarMenubar/ToolPageSpinner.cpp

```diff
@@ -41,19 +68,24 @@ auto ToolPageSpinner::newItem() -> GtkToolItem* {
     gtk_orientable_set_orientation(reinterpret_cast<GtkOrientable*>(spinner), orientation);
 
     GtkWidget* pageLabel = gtk_label_new(_("Page"));
-    this->lbPageNo = gtk_label_new(this->lbPageNo ? gtk_label_get_text(GTK_LABEL(lbPageNo)) : "");
+    this->lbPageNo = gtk_label_new("");
 
     if (horizontal) {
+        this->lbVerticalPdfPage = nullptr;
         gtk_widget_set_valign(pageLabel, GTK_ALIGN_BASELINE);
         gtk_widget_set_valign(spinner, GTK_ALIGN_BASELINE);
         gtk_widget_set_valign(this->lbPageNo, GTK_ALIGN_BASELINE);
     } else {
+        this->lbVerticalPdfPage = gtk_label_new("");
         gtk_widget_set_halign(pageLabel, GTK_ALIGN_BASELINE);
         gtk_widget_set_halign(spinner, GTK_ALIGN_BASELINE);
         gtk_widget_set_halign(this->lbPageNo, GTK_ALIGN_BASELINE);
+        gtk_widget_set_halign(lbVerticalPdfPage, GTK_ALIGN_BASELINE);
     }
 
-    GtkWidget* box = gtk_box_new(orientation, 1);
+    updateLabels();
+
+    box = gtk_box_new(orientation, 1);
```

Comment:
These two lines should be in reverse order. Otherwise, if ``box`` is null,  ``updateLabels()`` causes crash at ``gtk_box_pack_start`` in it. In addition, these two lines should be moved before the ``if-else clause`` just above them. Otherwise, parent widget checks in ``updateLabels()`` fails, and PDF page number won't shown.